.github/workflows/main-workflow.yaml

```diff
@@ -0,0 +1,43 @@
+name: Main-Workflow
+
+on:
+  push:
+    branches: [main, develop]
+  pull_request:
+    branches: [main, develop]
+
+
+jobs:
+  bump-dev-version: # only do that when actually merging in main/develop branch
+    if: github.event_name != 'pull_request'
+    uses: ./.github/workflows/bump-dev-version.yaml
+  R-CMD-Check:
+    if: ${{ always() }}
```

Comment:
Here and in other steps: why would we need this `if (true) `condition?

Reply:
All steps where we have `needs: [bump-dev-version]` would not be ran if `bump-dev-version` step is not executed (which is the case when we only push to the branch).

Reply:
ok, thx for the explanation :)